_studio/mfx_lib/fei/h264_enc/mfx_h264_enc.cpp

```diff
@@ -155,8 +155,8 @@ mfxStatus VideoENC_ENC::QueryStatus(DdiTask& task)
 
     for (mfxU32 f = f_start; f <= fieldCount; f++)
     {
-        sts = m_ddi->QueryStatus(task, task.m_fid[f]);
-        MFX_CHECK(sts != MFX_WRN_DEVICE_BUSY, MFX_TASK_BUSY);
+        while ((sts = m_ddi->QueryStatus(task, task.m_fid[f])) == MFX_WRN_DEVICE_BUSY)
+            vm_time_sleep(1);
```

Comment:
We are supposed to have blocking synchronization on Linux which the only one having FEI support. Why we can return MFX_WRN_DEVICE_BUSY at all?

Reply:
From MSDK side this status means that
```
        case VASurfaceRendering:
        case VASurfaceDisplaying:
            sts = MFX_WRN_DEVICE_BUSY;
```
where switch is performed on the returned value from
```
vaSts = vaQuerySurfaceStatus(m_vaDisplay, waitSurface, &surfSts);
```

According to https://github.com/Intel-Media-SDK/MediaSDK/issues/603 it can happen. I don't have any idea on beneath implementation or what is causing it

Reply:
You are not supposed to use vaQuerySurfaceStatus() at the first place. You **must** use vaSyncSurface(). Please, redesign the code and avoid artificial sleeps altogether. If there are bug in the driver or somewhere else preventing them - please, provide a list.

Reply:
@dvrogozh ok, I will redesign